File: laboratories/ppo/exercises/lab06/php/src/ParkingLot.php
<?php

namespace Ppo\Lab06;

use Ppo\Lab06\ParkingVehicles\ParkingVehicle;

class ParkingLot
{
    protected const MAX_AVAILABLE_SPACES = 3;
    protected int $occupiedSpaces = 0;

    public function letIn(ParkingVehicle $vehicle): void
    {
        $name = $vehicle->identify();
        $now = date("Y-m-d H:i:s");

        if ($this->occupiedSpaces >= static::MAX_AVAILABLE_SPACES) {
            echo "$name is returned at $now" . PHP_EOL;
            return;
        }

        echo "$name is entering at $now" . PHP_EOL;
        $this->occupiedSpaces++;
    }
}

File: laboratories/ppo/exercises/lab06/python/src/ParkingLot.py
from datetime import datetime
from src.parking_vehicles.ParkingVehicle import ParkingVehicle


class ParkingLot():
	MAX_AVAILABLE_SPACES = 3
	occupied_spaces = 0

	def let_in(self, vehicle):
		if not isinstance(vehicle, ParkingVehicle):
			raise Exception("object not implementing ParkingVehicle interface")

		name = vehicle.get_name()
		now = datetime.today().strftime('%Y-%m-%d %H:%M:%S')

		if self.occupied_spaces >= self.MAX_AVAILABLE_SPACES:
			print(name + " is returned at " + now)
			return

		print(name + " is entering at " + now)
		self.occupied_spaces = self.occupied_spaces + 1
